src/main/java/com/pnu/dev/radioserviceapi/service/VideoApiServiceImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.pnu.dev.radioserviceapi.service;
+
+import com.pnu.dev.radioserviceapi.client.YoutubeApiClient;
+import com.pnu.dev.radioserviceapi.client.dto.YoutubeApiResult;
+import com.pnu.dev.radioserviceapi.client.dto.search.YoutubeSearchResponse;
+import com.pnu.dev.radioserviceapi.dto.response.PageResponse;
+import com.pnu.dev.radioserviceapi.dto.response.VideoDto;
+import com.pnu.dev.radioserviceapi.dto.response.VideosCollectionResponse;
+import com.pnu.dev.radioserviceapi.exception.RadioServiceApiException;
+import com.pnu.dev.radioserviceapi.mongo.LiveBroadcastContent;
+import com.pnu.dev.radioserviceapi.mongo.Video;
+import com.pnu.dev.radioserviceapi.repository.VideoRepository;
+import com.pnu.dev.radioserviceapi.util.mapper.VideoMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class VideoApiServiceImpl implements VideoApiService {
+
+    private final YoutubeApiClient youtubeApiClient;
+
+    private final VideoRepository videoRepository;
+
+    private final VideoMapper videoMapper;
+
+    @Autowired
+    public VideoApiServiceImpl(YoutubeApiClient youtubeApiClient, VideoRepository videoRepository, VideoMapper videoMapper) {
+        this.youtubeApiClient = youtubeApiClient;
+        this.videoRepository = videoRepository;
+        this.videoMapper = videoMapper;
+    }
+
+    @Override
+    public PageResponse<VideoDto> findRecommended(Pageable pageable) {
+        pageable = PageRequest.of(pageable.getPageNumber(), pageable.getPageSize(), Sort.by("priority").ascending());
```

Comment:
Do not change pagable, it should be possible to pass `sort` to repository method as param